### DISCUSSION TITLE
How to disable auto-completion when some mode is used

### DISCUSSION THREAD

**Discussion by zikajk** [joaotavora/eglot#1194] [Q&A]
Hi, 
1. I am using CIder + Eglot + Corfu
2. Because Clojure-LSP is missing some capabilities in auto-completion I would like to have Cider manage auto-completion
3. What would be the correct way to disable Eglot's auto-completion (with pop-up) function when Cider is running?

Thank you

**Comment by zikajk** [comment]
This as a hook when cider starts seems to work...
```
  (defun my/remove-eglot-documentation-functions ()
            (remove-hook
             'eldoc-documentation-functions #'eglot-hover-eldoc-function t)
            (remove-hook              'eldoc-documentation-functions #'eglot-signature-eldoc-function t)
            (remove-hook              'eldoc-documentation-functions #'eglot-completion-at-point t))
```
Is it a good solution though?

**Reply by joaotavora** [reply]
Good one, depending on what cider uses you might also want to tweak `completion-at-point-functions`.

**Reply by joaotavora** [reply]
And you can put this hook in `eglot-managed-mode-hook`, to do it when Eglot starts instead. Should be roughly equivalent.

**Reply by zikajk** [reply]
I'm still solving the problem of how not to run it for files that are not attached to REPL, but that is not related to Eglot anymore. 
The` eglot-managed-mode-hook` seems to work little bit better, interesting... but maybe i will end up not using any hook and will run it interactively. Anyway, thanks!
```
(defun my/remove-eglot-documentation-functions ()
      (interactive)
      (when (bound-and-true-p cider-mode)
        (remove-hook 'eldoc-documentation-functions #'eglot-hover-eldoc-function t)
        (remove-hook 'eldoc-documentation-functions #'eglot-signature-eldoc-function t)
        (remove-hook 'eldoc-documentation-functions #'eglot-completion-at-point t)
        (setq completion-at-point-functions (remove 'eglot-completion-at-point completion-at-point-functions))))
```
